modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/dsl/Transaction.scala

```diff
@@ -174,9 +175,8 @@ object Transaction {
               case (fiber, Outcome.Canceled()) =>
                 fiber.cancel.timeout(timeouts.rollbackCommit)
             }
-            .adaptError {
-              case e: SQLException => new TransactionException(s"${e.getMessage} - SqlState: ${e.getSQLState}", e)
-              case e => new TransactionException(e.getMessage, e)
```

Comment:
Yes deliberately.  It interfered with the `isWorthRetry` in the `RetryingTransaction`, where I need to intercept and ignore errors that match the `ExceededRetriesException`.

You're right to pick up on this detail -- it's a bit too fragile at the moment.  One of the next things on my todo list is to improve how we create/wrap/unwrap exceptions.  And most importantly, how we report exceptions to the monitoring webhook.  But this will be in the next PR.